master/internal/api_experiment.go

```diff
@@ -351,70 +347,110 @@ func (a *apiServer) DeleteExperiment(
 func (a *apiServer) DeleteExperiments(
 	ctx context.Context, req *apiv1.DeleteExperimentsRequest,
 ) (*apiv1.DeleteExperimentsResponse, error) {
-	_, _, err := grpcutil.GetUser(ctx)
+	curUser, _, err := grpcutil.GetUser(ctx)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "failed to get the user: %s", err)
 	}
 
-	return &apiv1.DeleteExperimentsResponse{}, err
+	results, experiments, err := exputil.DeleteExperiments(ctx, a.m.system, req.ExperimentIds,
+		req.Filters)
+
+	go func() {
+		for i := 0; i < len(experiments); i += 10 {
+			exps := experiments[i : i+10]
+			if expIDs, err := a.deleteExperiments(context.Background(), exps, curUser); err != nil {
```

Comment:
q: how do this improve concurrency? `deleteExperiments` does a for loop to sequentially delete the list of 10 experiments, and for loop here sequentially runs `deleteExperiments` on these 10-element lists.